src/fidesops/service/connectors/saas_connector.py

```diff
@@ -100,47 +116,74 @@ def create_client(self) -> AuthenticatedClient:
         logger.info(f"Creating client to {uri}")
         return AuthenticatedClient(uri, self.configuration)
 
-    @staticmethod
-    def get_value_by_path(dictionary: Dict, path: str) -> Dict:
-        """Helper method to extract an arbitrary data path from a given dictionary"""
-        value = dictionary
-        for key in path.split("/"):
-            value = value[key]
-        return value
-
     def retrieve_data(
-        self, node: TraversalNode, policy: Policy, input_data: Dict[str, List[Any]]
+        self,
+        node: TraversalNode,
+        policy: Policy,
+        privacy_request: PrivacyRequest,
+        input_data: Dict[str, List[Any]],
     ) -> List[Row]:
         """Retrieve data from SaaS APIs"""
 
         # get the corresponding read request for the given collection
-        collection_name = node.address.collection
-        read_request = self.endpoints[collection_name].requests["read"]
+        collection_name: str = node.address.collection
+        read_request: SaaSRequest = self.endpoints[collection_name].requests["read"]
 
-        query_config = self.query_config(node)
+        query_config: SaaSQueryConfig = self.query_config(node)
         prepared_requests = query_config.generate_query(input_data, policy)
 
         rows: List[Row] = []
         for prepared_request in prepared_requests:
-            response = self.client().get(prepared_request)
+            response: Response = self.client().get(prepared_request)
+
+            if read_request.postprocessors is None:
+                rows.extend(response.json())
+                continue
+            data_to_be_processed: Any = self.post_process(
+                node.address,
+                privacy_request.get_cached_identity_data(),
+                read_request.postprocessors,
+                response,
+            )
+            rows.extend(data_to_be_processed)
```

Comment:
I'm wondering if there are scenarios where this processing could break the traversal - because we need to make sure this is always returning a list of Rows.  If the data we've unwrapped is an array of arrays for example - I think it would break downstream (for example GraphTask.pre_process_input_data consolidates data for inputting into downstream collections and expects lists of Rows, `filter_data_categories` at the end expects Rows.

Can we add a warning for this?

Reply:
Hmm so would this impl mean we need to loop through `data_to_be_processed` (both dict and list impl) and check if each element is a `Row`?

Reply:
I just worked with @galvana to do this:

```
if isinstance(data_to_be_processed, list):
                filter_only_dict = [item for item in data_to_be_processed if isinstance(item, dict)]
                if len(filter_only_dict) < len(data_to_be_processed):
                    logger.warning(f"Some data could not be added due to unexpected format")
                rows.extend(data_to_be_processed)
            elif isinstance(data_to_be_processed, dict):
                rows.append(data_to_be_processed)
            else:
                logger.warning(f"Some data could not be added due to unexpected format")
```

Reply:
nice solution-